opentracing/harness/api_check.py

```diff
@@ -88,10 +89,11 @@ def mock_finish(*_):
         else:
             raise AssertionError('Expected ValueError')  # pragma: no cover
 
-    def test_span_tags(self):
+    def test_span_tags_with_chaining(self):
         span = self.tracer().start_trace(operation_name='Farnsworth')
-        span.add_tag('birthday', '9 April, 2841')
-        span.add_tag('loves', 'different lengths of wires')
+        span. \
+            set_tag('birthday', '9 April, 2841'). \
+            set_tag('loves', 'different lengths of wires')
         span.finish()
```

Comment:
OK I think this is where I was going with in regards to inmem/mock etc. in
api-java, I'll certainly make assertions beyond compile not failing. while
implementations can throw certain things on the floor, we should be able to
write tests using keys that are expected to pass. In worst case, a tracer
can override and ignore the method if they throw everything on the floor.